test/sample-test/utils.py

```diff
@@ -59,4 +63,41 @@ def run_bash_command(cmd):
     output_string = output_bytes.decode('utf-8')
   if error_bytes != None:
     error_string = error_bytes.decode('utf-8')
-  return output_string, error_string
\ No newline at end of file
+  return output_string, error_string
+
+
+def file_injection(file_in, tmp_file_out, subs):
+  """Utility function that substitute several regex within a file by
+  corresponding string.
+
+  :param file_in: input file name.
+  :param tmp_file_out: tmp output file name.
+  :param subs: dict, key is the regex expr, value is the substituting string.
+  """
+  with open(file_in, 'rt') as fin:
+    with open(tmp_file_out, 'wt') as fout:
+      for line in fin:
+        tmp_line = line
+        for old, new in subs.items():
+          regex = re.compile(old)
+          tmp_line = re.sub(regex, new)
+
+        fout.write(tmp_line)
+
+  os.rename(tmp_file_out, file_in)
+
+def upload_blob(bucket_name, source_file_name, destination_blob_name):
+  """Uploads a file to the bucket."""
+  try:
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(bucket_name)
+    blob = bucket.blob(destination_blob_name)
+
+    blob.upload_from_filename(source_file_name)
+  except:
+    raise RuntimeError('Failure when uploading %s from %s to %s'
```

Comment:
The errors that GCS python throws might still help. For example, get_bucket throws bucket_not_found exception. What do you think of outputting this message before re-throwing the caught exception?

Reply:
Done.